evennia/contrib/tutorial_examples/tests.py

```diff
@@ -0,0 +1,69 @@
+from mock import Mock, patch
+
+from evennia.utils.test_resources import EvenniaTest
+
+from .bodyfunctions import BodyFunctions
+
+@patch("evennia.contrib.tutorial_examples.bodyfunctions.random")
+class TestBodyFunctions(EvenniaTest):
+    script_typeclass = BodyFunctions
+
+    def setUp(self):
+        super(TestBodyFunctions, self).setUp()
+        self.script.obj = self.char1
+
+    def tearDown(self):
+        super(TestBodyFunctions, self).tearDown()
+        # if we forget to stop the script, DirtyReactorAggregateError will be raised
+        self.script.stop()
+
+    def test_at_repeat(self, mock_random):
+        """test that no message will be sent when below the 66% threshold"""
+        mock_random.random = Mock(return_value=0.5)
+        old_func = self.script.send_random_message
+        self.script.send_random_message = Mock()
+        self.script.at_repeat()
+        self.script.send_random_message.assert_not_called()
+        # test that random message will be sent
+        mock_random.random = Mock(return_value=0.7)
+        self.script.at_repeat()
+        self.script.send_random_message.assert_called()
+        self.script.send_random_message = old_func
+
+    def test_send_random_message(self, mock_random):
```

Comment:
Wow, this is a highly detailed test of this return ... I would say (for future reference) that there is no need to test every individual string like this. It's thorough, but no real functionality will be safeguarded if the test fails because one of these strings are rephrased.